### PR TITLE
Fix: added original removed messaging providers doc page

### PR DESCRIPTION
**Pull request by ChiragAgg5k**
## What does this PR do?

This PR adds back the originally removed `docs/products/messaging/providers` documentation page. It was originally removed to make the providers more visible and easier to find, but some link according to issue #1479 still redirect to this page and having this page would still be beneficial.

## Test Plan

I have tested that the new page is rendering properly and updated the links to properly redirect to their respective documentations.

## Related PRs and Issues

* #1479 

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes I have.

Screenshot 2024-11-20 at 12 44 15 AM

### CODE REVIEW

**Comment by stnguyen90** [comment]
@ChiragAgg5k, next time, please make sure you ask to be assigned to the issue before submitting a PR.

File: src/routes/docs/products/messaging/providers/+page.markdoc [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+Deliver custom SMS messages to users using Twilio.
+{% /cards_item %}
+{% cards_item href="/docs/products/messaging/msg91" title="MSG91" icon="icon-msg91" %}
+Deliver custom SMS messages to users using MSG91.
+{% /cards_item %}
+{% cards_item href="/docs/products/messaging/telesign" title="Telesign" icon="icon-telesign" %}
+Deliver custom SMS messages to users using Telesign.
+{% /cards_item %}
+{% cards_item href="/docs/products/messaging/textmagic" title="TextMagic" icon="icon-textmagic" %}
+Deliver custom SMS messages to users using TextMagic.
```
Comment (stnguyen90) [inline src/routes/docs/products/messaging/providers/+page.markdoc:43]:
```suggestion
{% cards_item href="/docs/products/messaging/textmagic" title="Textmagic" icon="icon-textmagic" %}
Deliver custom SMS messages to users using Textmagic.
```

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
We also need the link added to the sidebar under Concepts:

https://github.com/appwrite/website/blob/f6f5b4bc834892bed6e76b2455f3c43af40058d7/src/routes/docs/products/messaging/%2Blayout.svelte#L23-L26

**Comment by ChiragAgg5k** [comment]
sorry about that 😅
will remember next time for sure

File: src/routes/docs/products/messaging/providers/+page.markdoc [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+{% /cards_item %}
+{% /cards %}
+# Email {% #email %}
+Deliver customized emails to users to send reminders, updates, promotions, and custom authentication logic.
+{% cards %}
+{% cards_item href="/docs/products/messaging/mailgun" title="Mailgun" icon="icon-mailgun" %}
+Deliver custom email messages to users using Mailgun.
+{% /cards_item %}
+{% cards_item href="/docs/products/messaging/sendgrid" title="Sendgrid" icon="icon-sendgrid" %}
+Deliver custom email messages to users using Sendgrid.
```
Comment (stnguyen90) [inline src/routes/docs/products/messaging/providers/+page.markdoc:27]:
```suggestion
{% cards_item href="/docs/products/messaging/sendgrid" title="SendGrid" icon="icon-sendgrid" %}
Deliver custom email messages to users using SendGrid.
```

**Comment by ChiragAgg5k** [comment]
added the fixes

**Comment by stnguyen90** [comment]
@ChiragAgg5k, please make sure to re-request a review when you're all done with your changes:

image

**Review by stnguyen90 — APPROVED** [review]

**Pull request merged**